tests/sentry/api/endpoints/test_project_stacktrace_link.py

```diff
@@ -225,6 +245,57 @@ def test_file_no_stack_root_match(self, mock_integration):
         assert response.data["error"] == "stack_root_mismatch"
         assert response.data["integrations"] == [serialized_integration(self.integration)]
 
+    @mock.patch.object(ExampleIntegration, "get_blame_for_file")
+    def test_get_commit_sha_valid_line_no(self, mock_blame):
+        mock_blame.return_value = git_blame
+        project_stacktrace_link_endpoint = ProjectStacktraceLinkEndpoint()
+        commit_sha = project_stacktrace_link_endpoint.get_commit_sha(
+            self.integration, self.organization.id, 26, "test/path/file.py", self.repo, "main"
+        )
+        assert commit_sha == "5c7dc040fe713f718193e28972b43db94e5097b4"
+
+    @mock.patch.object(ExampleIntegration, "get_blame_for_file")
+    def test_get_commit_sha_invalid_line_no(self, mock_blame):
+        mock_blame.return_value = git_blame
+        project_stacktrace_link_endpoint = ProjectStacktraceLinkEndpoint()
+        commit_sha = project_stacktrace_link_endpoint.get_commit_sha(
+            self.integration, self.organization.id, 80, "test/path/file.py", self.repo, "main"
+        )
+        assert commit_sha is None
+
+    @mock.patch("sentry.integrations.mixins.repositories.RepositoryMixin.get_stacktrace_link")
+    @mock.patch("sentry.integrations.github.client.GitHubClientMixin.get_blame_for_file")
+    @mock.patch("sentry.integrations.github.client.get_jwt", return_value=b"jwt_token_1")
+    @with_feature("organizations:codecov-stacktrace-integration")
+    def test_get_commit_sha_success(self, get_jwt, mock_git_blame, mock_get_stacktrace_link):
+        self.organization.flags.codecov_access = True
+        self.organization.save()
+        self.integration.provider = "github"
+        self.integration.save()
+
+        mock_get_stacktrace_link.side_effect = (
```

Comment:
I would expect a side effect to be an exception rather than a tuple. Could you please help me understand it?

Reply:
Sorry, this should have been `mock_get_stacktrace_link.return_value`. It is different from the other tests, as we are mocking `get_stacktrace_link` from `RepositoryMixin` instead of the `ExampleIntegration`. We need to do this since the provider is `githuub` instead of `example`.